packages/shared/friends/sagas.ts

```diff
@@ -2213,6 +2215,57 @@ export async function cancelFriendRequest(request: CancelFriendRequestPayload) {
   }
 }
 
+export async function rejectFriendRequest(request: RejectFriendRequestPayload) {
+  try {
+    // Get ownId value
+    const ownId = getOwnId(store.getState())
+    if (!ownId) {
+      return { reply: undefined, error: FriendshipErrorCode.FEC_UNKNOWN }
+    }
+
+    // Validate request
+    const isValid = validateFriendRequestId(request.friendRequestId, ownId)
+    if (!isValid) {
+      return { reply: undefined, error: FriendshipErrorCode.FEC_INVALID_REQUEST }
+    }
+
+    // Get otherUserId value
+    const userId = decodeFriendRequestId(request.friendRequestId, ownId)
+
+    // Search in the store for the message body
+    const messageBody = getMessageBody(store.getState(), userId)
+
+    // Update user data
+    store.dispatch(updateUserData(userId.toLowerCase(), getMatrixIdFromUser(userId)))
+
+    // Add as friend
+    const response = await UpdateFriendshipAsPromise(FriendshipAction.REJECTED, userId.toLowerCase(), false)
+
+    if (!response.error) {
+      const sendFriendRequest: RejectFriendRequestReplyOk = {
+        friendRequest: {
+          friendRequestId: request.friendRequestId,
+          timestamp: Date.now(),
+          from: userId,
+          to: getUserIdFromMatrix(ownId),
+          messageBody
+        }
+      }
+
+      // Return response
+      return { reply: sendFriendRequest, error: undefined }
+    } else {
+      // Return error
+      return { reply: undefined, error: response.error }
```

Comment:
How about having functions for `buildReply` and `buildError` to reuse everywhere instead of constructing objects everywhere?